Framework/API/src/Algorithm.cpp

```diff
@@ -735,6 +736,22 @@ bool Algorithm::executeInternal() {
     throw;
   }
 
+  catch (H5::Exception &ex) {
+
+    std::string errmsg;
+    errmsg.append(ex.getCFuncName()).append(": ").append(ex.getCDetailMsg());
+
+    m_runningAsync = false;
```

Comment:
From the other branches, it looks like this belongs as the first line in the error handling.

Reply:
Done